cypress/support/script_loader.js

```diff
@@ -124,10 +128,9 @@ function loadScriptsBeforeForUnitTests(...scriptKeys) {
     });
   }
   if (usesFirebase) {
-    // Currently no unit test actually writes to Firestore, they just use the
-    // library. A unit test that really writes to Firestore will be responsible
-    // for calling the necessary functions (addFirebaseHooks and logging in with
-    // the custom token, similar to what is done in prod).
+    // Some unit tests just use the Firebase library, so that's all we prepare
+    // for here. A unit test that really writes to Firestore has to call
+    // initFirebaseForUnitTest.
```

Comment:
why not always do initFirebaseForUnitTest? Are there a lot of tests where we use the script and don't actually read/write?

Reply:
Currently just unit_tests/polygon_draw_test. I thought about it, but this keeps this PR minimal in terms of functional change, and this does add some latency to each test case. I think that in a follow-up I'll have unit_tests/polygon_draw_test actually assert on the Firestore write contents. That might also allow us to delete some parts of integration_tests/polygon_draw_test with a clearer conscience

Reply:
sgtm!